pkg/container-utils/containerd/containerd.go

```diff
@@ -132,24 +132,19 @@ func (c *ContainerdClient) GetContainer(containerID string) (*runtimeclient.Cont
 		return nil, fmt.Errorf("listing labels of container %q: %w", container.ID(), err)
 	}
 
-	image, err := container.Image(c.ctx)
+	basicRuntimeMetadata, err := c.getBasicRuntimeMetadata(container, labels)
```

Comment:
What do you think about extending `getBasicRuntimeMetadata` to compute the whole `ContainerData`? It would need to:

- Receiving and setting the `state`
- Getting the labels by itself
- Calling `EnrichWithK8sMetadata`

It seems doable and the whole creation of `ContainerData` will be consolidated in one single place for all the cases.